common/src/main/java/com/revenuecat/purchases/common/BillingAbstract.kt

```diff
@@ -0,0 +1,97 @@
+package com.revenuecat.purchases.common
+
+import android.app.Activity
+import com.android.billingclient.api.BillingClient
+import com.android.billingclient.api.BillingResult
+import com.revenuecat.purchases.ProductType
+import com.revenuecat.purchases.PurchasesError
+import com.revenuecat.purchases.models.ProductDetails
+
+typealias ProductDetailsListCallback = (List<ProductDetails>) -> Unit
+
+typealias PurchasesErrorCallback = (PurchasesError) -> Unit
+
+abstract class BillingAbstract {
+
+    @get:Synchronized
+    @set:Synchronized
+    @Volatile
+    var stateListener: StateListener? = null
+
+    @get:Synchronized
+    @Volatile
+    var purchasesUpdatedListener: PurchasesUpdatedListener? = null
+        set(value) {
+            synchronized(this@BillingAbstract) {
+                field = value
+            }
+            if (value != null) {
+                startConnection()
+            } else {
+                endConnection()
+            }
+        }
+
+    interface StateListener {
+        fun onConnected()
+    }
+
+    abstract fun startConnection()
+
+    abstract fun endConnection()
```

Comment:
these used to be `private`, should they be `protected` now? otherwise they'll default to `public`, right?

Reply:
This class lives in the common module, and AmazonBilling and BillingWrapper live in their own module. If I make this protected (or internal in Kotlin), they won't be accessible for AmazonBilling and BillingWrapper and they won't be able to override them